component/src/main/java/org/cbioportal/genome_nexus/util/GenomicVariantUtil.java

```diff
@@ -138,6 +138,42 @@ public static boolean isMafFile(String file) {
         return Pattern.matches("^[\\w\\-\\./]+(\\.maf)|^[\\w\\-\\./]+(\\.tsv)|^[\\w\\-\\./]+(\\.txt)$$", file);
     }
 
+    // Hgvs formatted variants of several types are parsed for provided Reference Alleles
+    // empty string is returned if no reference allele can be extracted
+    public static String providedReferenceAlleleFromHgvs(String hgvs) {
+        // Substitutions require a single nucleotide reference sequence:
+        // 1g.123456A>G  (chromosome 1, genomic position 123456, reference sequence A, substituted with G)
+        // ref: http://varnomen.hgvs.org/recommendations/DNA/variant/substitution/
+        String match = getPattern("(?<=[^ATGC])[ATGC]>", hgvs);
+        if (match != null && match.trim().length() > 0) {
+            return match.substring(0, 1);
+        }
+        // Deletion-Insertions are recommended to *not* include specified deleted seqeunce/Reference Allele, but it is not prohibited.
+        // Examples showing specified deleted sequence/ReferenceAllele are mentioned here : http://varnomen.hgvs.org/recommendations/DNA/variant/delins/
+        // 1g.123456_123457delAAinsTT (chromosome 1, genomic positions 123456-123457, reference sequence AA deleted, sequence TT inserted in place)
+        // recommended representation omits the deleted nucleotides "1g.123456_123457delinsTT"
+        match = getPattern("del[ATGC]*ins", hgvs);
+        if (match != null && match.trim().length() > 5) {
```

Comment:
Yes, this is a little hacky. Here is the explanation:
This function makes three attempts to parse a ReferenceAllele string from HGVS. Each case "finds" the piece containing the reference allele with little flanking characters that need to be stripped. The idea behind the minimum length requirements was to avoid any chance of a IndexOutOfBoundsException getting thrown from String.substring(). In this middle case we are stripping 3 characters off of the start of the matched string and 3 characters off of the end of the matched string ... so we will be "safe" if we never attempt to strip from a string which is shorter than 6 characters. However, due to the nature of the regular expression, it is not possible for the matched string to be less than 6 characters so this check is not really needed here. Instead, we can be "safe" just by making sure the string "match" is not empty (length() > 0). I'll change to that here ... and the same goes for the stripping of 3 characters from the deletion case (see next comment)